subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ResultsStoreHelper.java

```diff
@@ -55,6 +55,10 @@ public static String determineChannel() {
         return System.getProperty(SYSPROP_PERFORMANCE_TEST_CHANNEL, "commits");
     }
 
+    public static Integer determineTeamCityBuildId() {
+        return System.getenv("BUILD_ID") == null ? 0 : Integer.parseInt(System.getenv("BUILD_ID"));
```

Comment:
We should decide on whether the build id can be null or not. I am not very fond of using `0` as a magic value for the build id, maybe `null` would be better.

How about we change the type of `BUILD_ID` to `String`, we actually don't care if it is an integer or not, right?

Reply:
Yes, buildId should be `String`: https://github.com/JetBrains/teamcity-rest-client/blob/master/teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt#L161